### ISSUE TITLE
DataTable: NumberFormatException in SortFeature.convertSortOrderParam

### ISSUE THREAD

**Issue by morvael** [primefaces/primefaces#4900]
## 1) Environment
- PrimeFaces version: **7.0.4**
- Application server + version: **Wildfly 16.0.0.Final**

## 2) Expected behavior
Clicking on header should sort data table.

## 3) Actual behavior
Exception is thrown: NumberFormatException: For input string: "NaN".

```
Caused by: java.lang.NumberFormatException: For input string: "NaN"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:76) [rt.jar:2.9 (01-24-2019)]
	at java.lang.Integer.parseInt(Integer.java:592) [rt.jar:2.9 (01-24-2019)]
	at java.lang.Integer.parseInt(Integer.java:627) [rt.jar:2.9 (01-24-2019)]
	at org.primefaces.component.datatable.feature.SortFeature.convertSortOrderParam(SortFeature.java:213) [primefaces-7.0.3.jar:7.0.3]
	at org.primefaces.component.datatable.feature.SortFeature.decode(SortFeature.java:65) [primefaces-7.0.3.jar:7.0.3]
	at org.primefaces.component.datatable.DataTableRenderer.decode(DataTableRenderer.java:64) [primefaces-7.0.3.jar:7.0.3]
	at javax.faces.component.UIComponentBase.decode(UIComponentBase.java:509) [jboss-jsf-api_2.3_spec-2.3.9.SP01.jar:2.3.9.SP01]
	at org.primefaces.component.api.UIData.processDecodes(UIData.java:311) [primefaces-7.0.3.jar:7.0.3]
	at com.sun.faces.context.PartialViewContextImpl$PhaseAwareVisitCallback.visit(PartialViewContextImpl.java:606) [jsf-impl-2.3.9.SP01.jar:2.3.9.SP01]
	at com.sun.faces.component.visit.PartialVisitContext.invokeVisitCallback(PartialVisitContext.java:159) [jsf-impl-2.3.9.SP01.jar:2.3.9.SP01]
	at org.primefaces.component.api.UIData.visitTree(UIData.java:866) [primefaces-7.0.3.jar:7.0.3]
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1468) [jboss-jsf-api_2.3_spec-2.3.9.SP01.jar:2.3.9.SP01]
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1468) [jboss-jsf-api_2.3_spec-2.3.9.SP01.jar:2.3.9.SP01]
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1468) [jboss-jsf-api_2.3_spec-2.3.9.SP01.jar:2.3.9.SP01]
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1468) [jboss-jsf-api_2.3_spec-2.3.9.SP01.jar:2.3.9.SP01]
	at javax.faces.component.UIForm.visitTree(UIForm.java:355) [jboss-jsf-api_2.3_spec-2.3.9.SP01.jar:2.3.9.SP01]
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1468) [jboss-jsf-api_2.3_spec-2.3.9.SP01.jar:2.3.9.SP01]
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1468) [jboss-jsf-api_2.3_spec-2.3.9.SP01.jar:2.3.9.SP01]
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1468) [jboss-jsf-api_2.3_spec-2.3.9.SP01.jar:2.3.9.SP01]
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1468) [jboss-jsf-api_2.3_spec-2.3.9.SP01.jar:2.3.9.SP01]
	at com.sun.faces.context.PartialViewContextImpl.processComponents(PartialViewContextImpl.java:400) [jsf-impl-2.3.9.SP01.jar:2.3.9.SP01]
	at com.sun.faces.context.PartialViewContextImpl.processPartial(PartialViewContextImpl.java:261) [jsf-impl-2.3.9.SP01.jar:2.3.9.SP01]
	... 69 more
```

Sorting works for colums where I have a `List<SortMeta>` initialized in backing bean (so they start ordered and change their order to descending/ascending when clicked). When I want to sort using previously unsorted column, I get the above exception. I think this function in `org.primefaces.component.datatable.feature.SortFeature`:
```java
    public String convertSortOrderParam(String order) {
        String sortOrder = null;
        int orderNumber = Integer.parseInt(order);

        switch (orderNumber) {
            case 0:
                sortOrder = "UNSORTED";
                break;

            case 1:
                sortOrder = "ASCENDING";
                break;

            case -1:
                sortOrder = "DESCENDING";
                break;
        }

        return sortOrder;
    }
```
should properly handle null or non-parseable order param. I don't know at which step "NaN" ends up encoded in `clientId + "_sortDir"` request parameter.

**Comment by morvael** [comment]
Suddenly I'm not getting that bug anymore. Perhaps it was present only on 7.0.3? I'm pretty sure I ran my tests with correct version of the library on development server. Now a fresh build with 7.0.4 on test server is no longer giving this issue.

**Comment by melloware** [comment]
I was going to mention I thought that was fixed in 7.0.4 but you had 7.0.4 in your description.  I think its definitely fixed.
